src/albert/resources/_mixins.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Generic, TypeVar
+
+from pydantic import PrivateAttr
+from typing_extensions import Self  # Python 3.10 fallback
+
+from albert.exceptions import AlbertException
+
+if TYPE_CHECKING:
+    from albert.collections.base import BaseCollection
+
+T = TypeVar("T")
+
+
+class HydrationMixin(Generic[T]):
+    _collection: BaseCollection | None = PrivateAttr(default=None)
```

Comment:
Should this inherit from `BaseAlbertModel` to define the `PrivateAttr` ? 

I am not sure how PrivateAttr works on class that does not inherit from Pydantic